app/scripts/modules/core/pipeline/config/templates/Variable.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import {IVariableMetadata} from './pipelineTemplate.service';
+import {variableInputService, IVariable, IVariableInput} from './variableInput.service';
+import autoBindMethods from 'class-autobind-decorator';
+
+interface IState { }
+
+interface IProps {
+  variableMetadata: IVariableMetadata;
+  variable: IVariable;
+  onChange: (variable: IVariable) => void;
+}
+
+@autoBindMethods
+export class Variable extends React.Component<IProps, IState> {
+
+  private getVariableInput(): JSX.Element {
+    const input: IVariableInput = variableInputService.getInputForType(this.props.variableMetadata.type);
+    return input ? input.getInput(this.props.variable, this.props.onChange) : null;
+  }
+
+  public render(): JSX.Element {
```

Comment:
We've discussed omitting the type declaration for `render()` methods, since it's always a `JSX.Element`. Feel free to omit.